syntax="proto2";
import "brpc/options.proto";
import "brpc/streaming_rpc_meta.proto";

package brpc.policy;
option java_package="com.brpc.policy";
option java_outer_classname="BaiduRpcProto";

message RpcMeta {
    optional RpcRequestMeta request = 1;
    optional RpcResponseMeta response = 2;
    optional int32 compress_type = 3;
    optional int64 correlation_id = 4;
    optional int32 attachment_size = 5;
    optional ChunkInfo chunk_info = 6;
    optional bytes authentication_data = 7;
    optional StreamSettings stream_settings = 8;   
}

message RpcRequestMeta {
    required string service_name = 1;
    required string method_name = 2;
    optional int64 log_id = 3;
    optional int64 trace_id = 4;
    optional int64 span_id = 5;
    optional int64 parent_span_id = 6;
}

message RpcResponseMeta {
    optional int32 error_code = 1;
    optional string error_text = 2;
}
